tests/test_versions.py

```diff
@@ -18,11 +18,24 @@ def test_addonversion_simple_major():
 
 def test_addonversion_kodi_versions():
     assert AddonVersion("1.0.0+matrix.1") > AddonVersion("1.0.0+leia.1")
+    assert AddonVersion("1.0.0+leia.1") == AddonVersion("1.0.0+Leia.1")
 
 
 def test_addonversion_kodi_simplevsstring():
     assert AddonVersion("0.5.4.1") > AddonVersion("0.5.4+matrix.1")
 
 
+def test_kodi_non_compliant_pep440_builtin_versions():
+    # mimics the tests in TestAddonVersion.cpp (non PEP440 compliant)
+    # try to use PEP440 compliant versions whenever possible
+    assert AddonVersion("1.0.0-1") > AddonVersion("1.0.0")
+    assert AddonVersion("1.0.0~beta2") > AddonVersion("1.0.0~beta1")
+    assert AddonVersion("1.0.1~beta01") > AddonVersion("1.0.0")
+    assert AddonVersion("1.0.1~beta") > AddonVersion("1.0.0")
+    assert AddonVersion("1.0.0~beta") > AddonVersion("1.0.0~alpha")
+    assert AddonVersion("1.0.0~beta2") > AddonVersion("1.0.0~alpha")
+    assert AddonVersion("1.0.0~alpha3") > AddonVersion("1.0.0~alpha2")
```

Comment:
Would tests with `+matrix.2` also make sense? Do we need to combine like `1.0.0+matrix.1~beta`?

Reply:
I don't think it makes much sense. `+matrix.1` is just a "wrapper" for the version available in matrix, so `1.0.0+matrix.1` is theoretically equivalent (at least for us) to version `1.0.0`. According to PEP440 is a local version identifier. 
Development versions in PEP440 are named `1.0.0betaX` or `1.0.0alphaX`.

So supposing that you have your `plugin.video.foo` in gotham (py3 compatible) with version 1.0.0:

version in gotham is `1.0.0`
version in matrix is `1.0.0+matrix.1`
version for development (external repo) is `1.0.1beta1`

`1.0.1beta1` > `1.0.0+matrix.1` > `1.0.0`

Reply:
This is a good reminder I need to update the wiki when this versioning stuff is finished